docs/dev/benchmark.rst

```diff
@@ -0,0 +1,116 @@
++-----------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------+------------------------------------------------------------------------------+
+| execution\_config     | {“number”: 1, “repeat”: 10, “min\_repeat\_ms”, 0}                                                                                                          | statistics   | workload execution parameters                                                |
++-----------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------+------------------------------------------------------------------------------+
+| statistics            | {“runtime\_ms\_mean”: 12,“runtime\_ms\_std”: 12}                                                                                                           | statistics   | all metrics, other optional fields ``binary_size``, ``compile_time``, etc.   |
++-----------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------+------------------------------------------------------------------------------+
+
+
+Storage format
+~~~~~~~~~~~~~~
+Currently we're prototyping benchmark data as JSON objects for extensibility and convenience, especially in early versions of the schema. However, as we scale up benchmark aggregation and stabilize parameters, we anticipate switching to a columnar format, such as Arrow or Parquet.
+
+Here is sample data encoded as JSON:
+
+::
+
+   {
+      "compilation_config":{
+         "relay_opt_lvl":3,
+         "tvm_target":"llvm"
+      },   
+      "engine":"tvm",
+      "engine_config":{
+         "llvm":"llvm-8"
+      },   
+      "engine_version":"3486e2c2cf9d0c7c20853f3503528444ce824c1b",
+      "hardware_config":"mbp13inch2019",
+      "metadata":{
+         "docker_tag":"tvmai/ci-gpu:v0.53"
+      },   
+      "schema_version":0.1,
+      "software_config":{
+         "os":"ubuntu:18.04",
+         "pip":{
+            "docker":"4.1.0",
+            "gitpython":"3.0.4",
+            "numpy":"1.17.4",
+            "onnx":"1.6.0"
+         }    
+      },   
+      "statistics":{
+         "compile_ms":2848.3831882476807,
+         "runtime_ms_mean":237.7542614444445,
+         "runtime_ms_std":2.8556173090926698
+      },   
+      "timestamp":"20191114064203",
+      "workload":"vgg19",
+      "workload_args":{
+         "input_name":"data_0",
+         "input_shape":[
+         1,   
+         3,   
+         224, 
+         224  
+         ]    
+      },   
+      "workload_metadata":{
+         "class":"vision",
```

Comment:
can we also add an "md5" field for the model sha? For this model, it would be `"md5":"07e9a6784c3d6d32e61e14ee8ceabfc7"`

Reply:
For general changes, we will need to start RFC again because it has not yet been discussed. Let us wait until it get merged, then add more info. Note that the we can add non-standard nested field contents

Reply:
@tqchen there is already a delta between the schema in this PR and the RFC you mention. For instance `runtime_config` is not listed anymore in this schema since there was redundancy with `software_config`. Should we re-spin the discussion in the RFC then?

Reply:
I think the main thing is no surprise to the community members. So if there is a change in the RFC, we should ideally respin the discussion and wait for another few days before we close

Reply:
Thanks Tianqi; I'll make sure any change to the log format is discussed in the RFC